simln-lib/src/lib.rs

```diff
@@ -995,18 +1020,17 @@ impl<C: Clock + 'static> Simulation<C> {
             active_nodes.insert(node_info.pubkey, (node_info, capacity));
         }
 
-        // Create a network generator with a shared RNG for all nodes.
-        let network_generator = Arc::new(Mutex::new(
+        // in order to choose a deterministic destination is necessary to sort the nodes by its public key
+        let mut sorted_actives_nodes: Vec<(NodeInfo, u64)> =
+            active_nodes.values().cloned().collect();
+        sorted_actives_nodes.sort_by(|n1, n2| n1.0.pubkey.cmp(&n2.0.pubkey));
+
+        let network_generator = Arc::new(
             NetworkGraphView::new(
-                active_nodes.values().cloned().collect(),
+                sorted_actives_nodes,
```

Comment:
Here the caller somewhat assumes understanding of the implementation of the network generator. 

I think that we should move this sort into `NetworkGraphView` and then add a note on the trait that implementations should be deterministic if called with the same seed.